Apps/Admin/Tests/Functional/cypress/integration/e2e/patient-details.cy.js

```diff
@@ -59,6 +59,10 @@ describe("Patient details message verification", () => {
             .click();
 
         cy.get("[data-testid=block-access-switches]").should("be.visible");
+        cy.get(`[data-testid*="block-access-switch"]`).should(
+            "not.have.attr",
+            "readonly"
+        );
         cy.get("[data-testid=block-access-cancel]").should("not.exist");
```

Comment:
The cancel and save checks will always be 'not.exist' unless a dataset item has been selected i.e., they do not show up until a dataset is selected.   You need to select a dataset before checking if buttons exist or not.